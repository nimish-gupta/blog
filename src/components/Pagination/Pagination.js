// @flow strict
import React from 'react';

import { PAGINATION } from '../../constants';
import * as Styled from './Pagination.style';

type Props = {
	prevPagePath: string,
	nextPagePath: string,
	hasNextPage: boolean,
	hasPrevPage: boolean,
};

const Pagination = ({
	prevPagePath,
	nextPagePath,
	hasNextPage,
	hasPrevPage,
}: Props) => {
	const NextLink = hasNextPage ? Styled.NavLink : Styled.DisabledNavLink;
	const PrevLink = hasPrevPage ? Styled.NavLink : Styled.DisabledNavLink;

	return (
		<Styled.Pagination>
			<Styled.LinkContainer align="left">
				<PrevLink rel="prev" to={hasPrevPage ? prevPagePath : '/'}>
					{PAGINATION.PREV_PAGE}
				</PrevLink>
			</Styled.LinkContainer>
			<Styled.LinkContainer align="right">
				<NextLink rel="next" to={hasNextPage ? nextPagePath : '/'}>
					{PAGINATION.NEXT_PAGE}
				</NextLink>
			</Styled.LinkContainer>
		</Styled.Pagination>
	);
};

export default Pagination;
