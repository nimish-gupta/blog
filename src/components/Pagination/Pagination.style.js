import styled from 'styled-components';
import { lighten } from 'polished';
import { Link } from 'gatsby';

import { marginTop } from '../../assets/styled-components/mixins';

export const Pagination = styled.div`
	${marginTop(2)};
	display: flex;
`;

export const LinkContainer = styled.div`
	width: 50%;
	text-align: ${(props) => props.align};
`;

export const NavLink = styled(Link)`
	color: ${(props) => props.theme.color.secondary};
	font-size: 26px;
	font-weight: bold;

	&:hover,
	&:focus {
		color: ${(props) => props.theme.color.primary};
	}
`;

export const DisabledNavLink = styled(NavLink)`
	pointer-events: none;
	color: ${(props) => lighten(0.2, props.theme.color.gray)};
`;
