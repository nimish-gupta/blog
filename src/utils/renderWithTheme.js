import React from 'react';
import renderer from 'react-test-renderer';
import { ThemeProvider } from 'styled-components';
import getTheme from '../assets/styled-components/_variables';

export default function renderWithTheme(component) {
	return renderer.create(
		<ThemeProvider theme={getTheme('light')}>{component}</ThemeProvider>
	);
}
