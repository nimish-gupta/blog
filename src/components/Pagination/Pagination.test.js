// @flow strict
import React from 'react';

import Pagination from './Pagination';
import renderWithTheme from '../../utils/renderWithTheme';

describe('Pagination', () => {
	const props = {
		prevPagePath: '/page/1',
		nextPagePath: '/page/3',
		hasNextPage: true,
		hasPrevPage: true,
	};

	it('renders correctly', () => {
		const tree = renderWithTheme(<Pagination {...props} />).toJSON();
		expect(tree).toMatchSnapshot();
	});
});
