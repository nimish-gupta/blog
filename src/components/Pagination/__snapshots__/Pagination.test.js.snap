// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pagination renders correctly 1`] = `
<div
  className="sc-AxjAm gYaWGF"
>
  <div
    className="sc-AxirZ jJzPOj"
  >
    <a
      className="sc-AxiKw buROYx"
      href="/page/1"
      rel="prev"
    >
      ← PREV
    </a>
  </div>
  <div
    className="sc-AxirZ ucxGs"
  >
    <a
      className="sc-AxiKw buROYx"
      href="/page/3"
      rel="next"
    >
      → NEXT
    </a>
  </div>
</div>
`;
